forte/data/data_pack.py

```diff
@@ -387,6 +387,10 @@ def delete_entry(self, entry: EntryType):
                 f"should be an instance of Annotation, Link, or Group."
             )
 
+        if not isinstance(entry, Link):
+            logger.warning("Deleting an annotation or a group doesn't "
+                           "guarantee deletion of the associated links.")
+
```

Comment:
Yes, makes sense.
I'm actually myself not sure what would happen. In case of annotations we are removing the annotation from `self.annotation` list. Does that mean null reference won't happen? Because the annotation object still exists.
Also, I don't know what a group is. Can we discuss this today?

Reply:
A `link` is used to link two entries. And a `group` is used to cluster/group more than 2 entries, for example, groups can be used to represent all students in the same class.

Here you can simply say, the links and groups referencing to by these object still exists, one need to take care of them to avoid unexpected behavior (mostly they will find `None`).

Reply:
Sure, added the explanation to the warning.